core/modules/search/search.module

```diff
@@ -1422,3 +1422,61 @@ function search_config_info() {
   );
   return $prefixes;
 }
+
+/**
+ * Batch operation to rebuild the search index.
+ * 
+ * @see search_reindex_confirm()
+ * @see search_reindex_confirm_submit()
+ */
+function search_reindex_batch($passes_needed, $search_active_modules, &$context) {
+  if (!isset($context['sandbox']['current_pass'])) {
+    $context['sandbox']['current_pass'] = 0;
+  }
+
+  // Invoke hook_update_index for all search active modules. 
+  if ($context['sandbox']['current_pass'] < $passes_needed) {
+    foreach ($search_active_modules as $module) {
+      module_invoke($module, 'update_index');
+    }
+    $context['sandbox']['current_pass']++;
+    $context['message'] = t('Now processing pass %pass of %needed', array('%pass' => $context['sandbox']['current_pass'] + 1, '%needed' => $passes_needed));
+    $context['finished'] = $context['sandbox']['current_pass'] / $passes_needed;
+  }
+}
+
+/**
+ * Batch 'finished' callback for rebuilding the search index.
+ * 
+ * @see search_reindex_batch()
+ */
+function search_reindex_batch_finished($success, $results, $operations) {
+  if ($success) {
+    $config = config('search.settings');
+    // Collect some stats
+    $remaining = 0;
+    $total = 0;
+    foreach ($config->get('search_active_modules') as $module) {
+      if ($status = module_invoke($module, 'search_status')) {
+        $remaining += $status['remaining'];
+        $total += $status['total'];
+      }
+    }
```

Comment:
This same bit of code is in `search_admin_settings()` and something similar in `search_reindex_confirm_submit()` above. Could we put this into a new API function? Maybe named `search_get_status($module = NULL)` with a parameter for which module to check, or NULL for all modules like this.